src/api/kube.ts

```diff
@@ -555,6 +555,20 @@ export class KubeHelper {
     }
   }
 
+  async hasReadPermissionsForNamespace(namespace: string): Promise<boolean> {
```

Comment:
k8s provides for us api to access estimation. Could you use it? :

Proposed changes:

```typescript
    const k8sApi = KubeHelper.KUBE_CONFIG.makeApiClient(AuthorizationV1Api)
    const review = new V1SelfSubjectAccessReview()
    review.spec = {}
    review.spec.resourceAttributes = {
      group: "", 
      name: "che-access-to-namespace", 
      namespace, 
      resource: "namespaces", 
      verb: "get"
    }

    try {
      const { body } = await k8sApi.createSelfSubjectAccessReview(review)
      return body.status!.allowed
    } catch (error) {
      if (error.response && error.response.body) {
        if (error.response.body.code === 403) {
          return false
        }
      }
      throw this.wrapK8sClientError(error)
    }

```
For me it looks very similar to this one pr: https://github.com/eclipse/che-operator/pull/209